components/Templates/includes/auto-template/Pods_Templates_Auto_Template_Front_End.php

```diff
@@ -0,0 +1,368 @@
+
+				//if auto template is enabled add info about Pod to array
+				if ( 1 == pods_v( 'pfat_enable', $pods->pod_data[ 'options' ] ) ) {
+					//check if pfat_single and pfat_archive are set
+					$single = pods_v( 'pfat_single', $pods->pod_data[ 'options' ], false, true );
+					$archive = pods_v( 'pfat_archive', $pods->pod_data[ 'options' ], false, true );
+					$single_append = pods_v( 'pfat_append_single', $pods->pod_data[ 'options' ], true, true );
+					$archive_append = pods_v( 'pfat_append_archive', $pods->pod_data[ 'options' ], true, true );
+					$type = pods_v( 'object_type', $pods->pod_data, false, true );
+					//check if it's a post type that has an arhive
+					if ( $pods->pod_data['type'] === 'post_type' && $the_pod !== 'post' || $the_pod !== 'page' ) {
+						$has_archive = pods_v( 'has_archive', $pods->pod_data['options'], false, true );
+					}
+					else {
+						$has_archive = true;
+					}
+
+					//build output array
+					$auto_pods[ $the_pod ] = array(
+						'name' => $the_pod,
+						'label'	=> $the_pod_label,
+						'single' => $single,
+						'archive' => $archive,
+						'single_append' => $single_append,
+						'archive_append' => $archive_append,
+						'has_archive'	=> $has_archive,
+						'type' => $type,
+					);
+				}
+
+			} //endforeach
+
+			//cache the results
+			pods_transient_set( $key, $auto_pods );
+		}
+
+		/**
+		 * Add to or change settings.
+		 *
+		 * Use this filter to change or add to the settings set in the back-end for this plugin. Has no effect if 'pods_pfat_auto_pods_override' filter is being used.
+		 *
+		 * @param array $auto_pods Array of parameters to use instead of those from settings.
+		 *
+		 * @return array Settings arrays for each post type.
+		 *
+		 * @since 2.4.5
+		 */
+		$auto_pods = apply_filters( 'pods_pfat_auto_pods', $auto_pods );
+
+		return $auto_pods;
+
+	}
+
+	/**
+	 * Fetches the current post type.
+	 *
+	 * @return string current post type.
+	 *
+	 * @since 2.4.5
+	 */
+	function current_post_type() {
+		//start by getting current post or stdClass object
+		global $wp_query;
+		$obj = $wp_query->get_queried_object();
+
+		//see if we are on a post type and if so, set $current_post_type to post type
+		if ( isset( $obj->post_type ) ) {
+			$current_post_type = $obj->post_type;
+
+		}
+		elseif ( isset( $obj->taxonomy ) ) {
+			$current_post_type = $obj->taxonomy;
+		}
+		elseif ( isset ( $obj->name ) ) {
+			$current_post_type = $obj->name;
+		}
+		elseif ( is_home() ) {
+			$current_post_type = 'post';
+		}
+		else {
+			$current_post_type = false;
+		}
+
+		return $current_post_type;
+	}
+
+	/**
+	 * Outputs templates after the content as needed.
+	 *
+	 * @param string $content Post content
+	 *
+	 * @uses 'the_content' filter
+	 *
+	 * @return string Post content with the template appended if appropriate.
+	 *
+	 * @since 2.4.5
+	 */
+	function front( $content ) {
+
+		// cet the current post type
+		$current_post_type = $this->current_post_type();
```

Comment:
@josh Is current_post_type() right for this? It checks current queried object but that's in WP_Query, that's not the current post data set on the screen.

We're running this on a post content / excerpt filter, we should just pull `get_post_type()`, and support a second parameter (`the_content` default passes in `$post_id` as second parameter in some cases). Also, if we add support for the optional second `$post_id = null` parameter in front(), we should be sure to update our usage of add_filter / remove_filter to include priority: 10 / arguments: 2 otherwise we'll only ever get the `$content` from the filter.